apps/finance/app/src/components/Transfers.js

```diff
@@ -161,143 +158,184 @@ const Transfers = React.memo(({ dao, tokens, transactions }) => {
             `}
           >
-            {filteredTransfers
-              .sort(({ date: dateLeft }, { date: dateRight }) =>
-                // Sort by date descending
-                compareDesc(dateLeft, dateRight)
-              )
-              .slice(0, displayedTransfers)
-              .map(transfer => (
-                <TransferRow
-                  key={transfer.transactionHash}
-                  token={tokenDetails[toChecksumAddress(transfer.token)]}
-                  transaction={transfer}
-                  smallViewMode={compactMode}
-                />
-              ))}
-          </Table>
-          {displayedTransfers < filteredTransfers.length && (
-            <Footer compactMode={compactMode}>
-              <Button mode="secondary" onClick={showMoreTransfers}>
-                Show Older Transfers
+            <div
+              css={`
+                color: ${theme.content};
+                ${textStyle('body1')}
+              `}
+            >
+              Transfers
+            </div>
+            <div css="text-align: right;">
+              <Button onClick={handleDownload}>
+                <IconExternal /> Export
               </Button>
-            </Footer>
+            </div>
+          </div>
+          {layoutName !== 'small' && (
+            <TransfersFilters
+              dateRangeFilter={selectedDateRange}
+              onDateRangeChange={handleSelectedDateRangeChange}
+              tokenFilter={selectedToken}
+              onTokenChange={handleTokenChange}
+              transferTypeFilter={selectedTransferType}
+              onTransferTypeChange={handleTransferTypeChange}
+              compactMode={compactMode}
+              symbols={symbols}
+              transferTypes={TRANSFER_TYPES_STRING}
+            />
           )}
-        </div>
+          {emptyResultsViaFilters && (
+            <EmptyFilteredTransfers onClear={handleClearFilters} />
+          )}
+        </React.Fragment>
+      }
+      fields={
+        emptyResultsViaFilters
+          ? []
+          : [
+              { label: 'Date', priority: 2 },
+              { label: 'Source/recipient', priority: 3 },
+              { label: 'Reference', priority: 1 },
+              { label: 'Amount', priority: 2 },
+            ]
+      }
+      entries={filteredTransfers.sort(
+        ({ date: dateLeft }, { date: dateRight }) =>
+          // Sort by date descending
+          compareDesc(dateLeft, dateRight)
+      )}
+      renderEntry={({
+        date,
+        entity,
+        reference,
+        isIncoming,
+        numData: { amount },
+        token,
+      }) => {
+        const formattedDate = format(date, "yyyy-MM-dd'T'HH:mm:ss.SSSxxx")
+        const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
+        const formattedAmount = formatTokenAmount(
+          amount,
+          isIncoming,
+          decimals,
+          true,
+          { rounding: 5 }
+        )
+
+        return [
```

Comment:
It's nice that we don't have to use a fragment, but the array syntax always scares me a bit (always hard to figure out where to put the `,`, and it's harder to read).

Does it work if we wrap it in a fragment?

Reply:
It does not work that way (correct me if I'm wrong here @bpierre) I share the concern but this is a way to set the content per each column.

Reply:
Yes this is an array of separated nodes that correspond to each field, and can be rendered in the context of a table row or a list block.

We could also accept an object here, using the field labels as IDs:

```jsx
<DataView
  renderEntry={entry => {
    return {
      'Date': <Date />,
      'Source/recipient': <Source />,
      'Reference': <Reference />,
      'Amount': <Amount />,
    }
  }}
/>
```

We could also have an optional ID on the fields, to make it a bit nicer:

```jsx
<DataView
  fields={[
    { id: 'date', label: 'Date' },
    { id: 'source', label: 'Source/recipient' },
    { id: 'reference', label: 'Reference' },
    { id: 'amount', label: 'Amount' },
  ]}
  renderEntry={entry => {
    return {
      date: <Date />,
      source: <Source />,
      reference: <Reference />,
      amount: <Amount />,
    }
  }}
/>
```

But having optional IDs might bring too much confusion for users and I would rather make them mandatory in that case… which would also make simple cases more verbose by removing the possibility to use strings directly for fields.

Reply:
Actually we could accept both styles: a function returning an array of elements, or an array of components.

Reply:
I was just thinking of another option so I’m adding it here as well.

We could be to move these render functions / components into `fields`:

```jsx
<DataView
  fields={[
    { label: 'Date', render: Date },
    { label: 'Source/recipient', render: Source },
    { label: 'Reference', render: Reference },
    { label: 'Amount', render: Amount },
  ]}
/>
```

And we could still have a concise mode:

```jsx
<DataView
  fields={[
    [ 'Date', ({ entry }) => entry.date ],
    [ 'Account', ({ entry }) => entry.account ],
  ]}
/>
```

Now it would be a bit weird to have `renderEntryActions` and `renderEntryChild` separated from the rest, so maybe we could also move `fields`, `renderEntryActions` and `renderEntryChild` under a single `entry` (or `entryConfig`) prop, containing everything related to the configuration of individual entries:

```jsx
<DataView
  entry={
    fields: [
      { 'Date', render: Date },
      { label: 'Source/recipient', render: Source },
      { label: 'Reference', render: Reference },
      { label: 'Amount', render: Amount },
    ],
    actions: () => null,
    child: () => null,
  }
/>
```

The concise mode would look like this:

```jsx
<DataView
  entry={
    fields: [
      [ 'Date', ({ entry }) => entry.date ],
      [ 'Account', ({ entry }) => entry.account ],
    ]
  }
/>
```

And we could even allow to skip `fields` if there is no action or child content:

```jsx
<DataView
  entry={[
    [ 'Date', ({ entry }) => entry.date ],
    [ 'Account', ({ entry }) => entry.account ],
  ]}
/>
```

Reply:
I really like this last option, I think it makes more sense to put these together.

Reply:
Maybe identity as default?

Reply:
Moved to https://github.com/aragon/aragon-ui/issues/468.